### ISSUE TITLE
"downloadedTypeError: Cannot read property 'response' of undefined" when trying to lint certain URLs

### ISSUE THREAD

**Issue by pdehaan** [sonarwhal/sonar#324]
I have **sonar** (great tool, btw!!) installed locally, and it works great for a bunch of the sites I've tried it on, but then for this random URL, it seems to be throwing some unhandled `downloadedTypeError` and I believe skipping some rules:

```sh
$(npm bin)/sonar "https://fileshare.stage.mozaws.net/download/7639359c2007fb9c5403018fe4cc0672/#lNCdQX1SYfJaQudAk8C52g"

| https://fileshare.stage.mozaws.net/bundle.js downloadedTypeError: Cannot read property 'response' of undefined
    at CDPCollector.get headers [as headers] (/Users/pdehaan/dev/tmp/del/sonar-test/node_modules/@sonarwhal/sonar/dist/src/lib/collectors/cdp/cdp.js:536:39)
    at Sonar.get pageHeaders [as pageHeaders] (/Users/pdehaan/dev/tmp/del/sonar-test/node_modules/@sonarwhal/sonar/dist/src/lib/sonar.js:156:30)
    at RuleContext.get pageHeaders [as pageHeaders] (/Users/pdehaan/dev/tmp/del/sonar-test/node_modules/@sonarwhal/sonar/dist/src/lib/rule-context.js:31:26)
    at validate (/Users/pdehaan/dev/tmp/del/sonar-test/node_modules/@sonarwhal/sonar/dist/src/lib/rules/highest-available-document-mode/highest-available-document-mode.js:122:46)
    at Immediate.setImmediate [as _onImmediate] (/Users/pdehaan/dev/tmp/del/sonar-test/node_modules/@sonarwhal/sonar/dist/src/lib/sonar.js:109:50)
    at runCallback (timers.js:800:20)
    at tryOnImmediate (timers.js:762:5)
    at processImmediate [as _immediateCallback] (timers.js:733:5)
(node:82972) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 4): TypeError: Cannot read property 'response' of undefined
(node:82972) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
✖ Finishing...
https://fileshare.stage.mozaws.net/bundle.js
line 3  col 3  Error  'content-type' header should have 'charset=utf-8'  content-type
✖ Found 1 error and 0 warnings

https://fileshare.stage.mozaws.net/dow … 403018fe4cc0672/#lNCdQX1SYfJaQudAk8C52g
line 1   col 16  Error  <html> element must have a lang attribute     axe
line 17  col 11  Error  Elements must have sufficient color contrast  axe
line 32  col 7   Error  Elements must have sufficient color contrast  axe
✖ Found 3 errors and 0 warnings

✖ Found a total of 4 errors and 0 warnings
Exit code: 1
```

And my config is:

```
{
  "browserslist": "",
  "collector": {
    "name": "cdp",
    "options": {}
  },
  "formatter": "stylish",
  "ignoredUrls": {
    "code.cdn.mozilla.net": ["*"]
  },
  "rules": {
    "axe": "error",
    "content-type": "error",
    "disallowed-headers": "error",
    "disown-opener": "error",
    "highest-available-document-mode": "error",
    "manifest-exists": "off",
    "manifest-file-extension": "error",
    "manifest-is-valid": "error",
    "meta-charset-utf-8": "error",
    "no-friendly-error-pages": "error",
    "no-html-only-headers": "off",
    "no-protocol-relative-urls": "error",
    "ssllabs": "error",
    "x-content-type-options": "error"
  },
  "rulesTimeout": 120000
}
```

**Comment by molant** [comment]
I just tried with the url in the top of your output and I'm having problems too:
```
rror: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/factoryregistry.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/origincheck.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/webrequestsender.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/requesthelper.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/signer.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/requestqueue.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/devicestatuscodes.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/googlecorpindividualattest.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/closeable.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/watchdog.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/gnubby.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/gnubbies.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/webrequest.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/usbsignhandler.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/logging.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/messagetypes.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/gnubby-u2f.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/approvedorigins.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/usbenrollhandler.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/generichelper.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/inherits.js
Error: chrome-extension://kmendfapggjehodndflmmgagdbamhnfd/errorcodes.js
```

I think this is something similar to #274 (at least in my output).

@sarvaje can you take a look at this and maybe put it as part of the refactor of CDP (#297)

**Comment by pdehaan** [comment]
Ah, yes... If I switch from the `cdp` [collector](https://sonarwhal.com/docs/user-guide/collectors/index.html) to `jsdom`, the error seems to go away:

```sh
> $(npm bin)/sonar "https://testpilot.firefox.com"

✖ Finishing...
https://testpilot.firefox.com/
                  Error  'x-ua-compatible' header was not specified  highest-available-document-mode
line 1  col 1     Error  <html> element must have a lang attribute   axe
line 5  col 2340  Error  id attribute value must be unique           axe
✖ Found 3 errors and 0 warnings

✖ Found a total of 3 errors and 0 warnings
Exit code: 1
```

```sh
> $(npm bin)/sonar "https://fileshare.stage.mozaws.net/download/7639359c2007fb9c5403018fe4cc0672/#lNCdQX1SYfJaQudAk8C52g"

✖ Finishing...
https://fileshare.stage.mozaws.net/dow … 403018fe4cc0672/#lNCdQX1SYfJaQudAk8C52g
               Error  'x-ua-compatible' header was not specified  highest-available-document-mode
               Error  No charset meta tag was specified           meta-charset-utf-8
line 1  col 1  Error  <html> element must have a lang attribute   axe
✖ Found 3 errors and 0 warnings

✖ Found a total of 3 errors and 0 warnings
Exit code: 1
```

```sh
> cat .sonarrc
{
  "browserslist": "",
  "collector": {
    "name": "jsdom",
    "options": {}
  },
  "formatter": "stylish",
  "ignoredUrls": {
    "code.cdn.mozilla.net": ["*"],
    "analysis-output.telemetry.mozilla.org": ["*"],
    "ssl.google-analytics.com": ["*"]
  },
  "rules": {
    "axe": "error",
    "content-type": "off",
    "disallowed-headers": "error",
    "disown-opener": "error",
    "highest-available-document-mode": "error",
    "manifest-exists": "off",
    "manifest-file-extension": "error",
    "manifest-is-valid": "error",
    "meta-charset-utf-8": "error",
    "no-friendly-error-pages": "error",
    "no-html-only-headers": "off",
    "no-protocol-relative-urls": "error",
    "ssllabs": "error",
    "x-content-type-options": "error"
  },
  "rulesTimeout": 120000
}
```

**Comment by sarvaje** [comment]
Hi @pdehaan,

I've been taking a look, and the problem with https://fileshare.stage.mozaws.net/download/7639359c2007fb9c5403018fe4cc0672/#lNCdQX1SYfJaQudAk8C52g is because of the hash, but it should be fixed in #331 

The error with https://testpilot.firefox.com is related with #275

**Comment by sarvaje** [comment]
@pdehaan you can find a workaround for the issue with https://testpilot.firefox.com in #275
